service/frontend/workflowHandler.go

```diff
@@ -722,18 +727,24 @@ func (wh *WorkflowHandler) RecordActivityTaskHeartbeat(
 		return nil, wh.error(err, scope)
 	}
 
+	domainWrapper := domainWrapper{
+		domain: domainName,
+	}
 	scope, sw := wh.startRequestProfileWithDomain(
 		ctx,
 		metrics.FrontendRecordActivityTaskHeartbeatScope,
-		domainWrapper{
-			domain: domainName,
-		},
+		domainWrapper,
 	)
 	defer sw.Stop()
 
-	if wh.isShuttingDown() {
-		return nil, errShuttingDown
-	}
+	// Count the request in the RPS towards the requested domain,
+	// but we still accept it even if RPS is exceeded
+	wh.allow(domainWrapper)
```

Comment:
previously this `wh.allow(nil)` which won't consume RPS token for the requested domain, only token for the global RPS is consumed.